src/deephaven/plugin/object.py

```diff
@@ -17,46 +17,84 @@ def index(self) -> int:
         May be used in the output stream to refer to the reference from the client."""
         return self._index
 
-    @property
-    def type(self) -> Optional[str]:
-        """The type."""
-        return self._type
-
 
 class Exporter(abc.ABC):
-    """The interface for creating new references during ObjectType.to_bytes."""
+    """The interface for creating new references during FetchOnlyObjectBase.to_bytes."""
+
+    @abc.abstractmethod
+    def reference(self, obj: object) -> Reference:
+        """Creates a reference for an object. Each reference """
+        pass
+
+
+class MessageStream(abc.ABC):
+    """A stream of messages, either sent from server to client or client to server. ObjectType implementations
+    provide an instance of this interface for each incoming stream to invoke as messages arrive, and will
+    likewise be given an instance of this interface to be able to send messages to the client.
+    """
+    def __init__(self):
+        pass
 
     @abc.abstractmethod
-    def reference(
-        self, object, allow_unknown_type: bool = False, force_new: bool = False
-    ) -> Optional[Reference]:
-        """Gets the reference for object if it has already been created and force_new is False,
-        otherwise creates a new one. If allow_unknown_type is False, and no type can be found, no
-        reference will be created."""
+    def on_close(self):
```

Comment:
Should this be noted as an abstractmethod, or is it ok not to force implementation?

Reply:
It is technically not wrong to do nothing, but for a bidirectional plugin (which this must be), omitting this means not knowing when to stop sending responses (and each subsequent message sent will result in an error), potentially free resources.

Reply:
@jmao-denver